client/signup/steps/site-topic/index.jsx

```diff
@@ -26,23 +28,36 @@ class SiteTopicStep extends Component {
 		setSiteTopic: PropTypes.func.isRequired,
 		signupProgress: PropTypes.array,
 		stepName: PropTypes.string,
+		siteTopic: PropTypes.string,
 	};
 
 	state = {
-		siteTopic: '',
+		siteTopicInputValue: '',
 	};
 
-	onChangeTopic = event => this.setState( { siteTopic: event.target.value } );
+	componentDidMount() {
+		const { siteTopic } = this.props;
 
-	// TODO:
-	// Handle submission.
-	submitSiteTopic( event ) {
-		event.preventDefault();
+		if ( siteTopic ) {
+			this.state.siteTopicInputValue = siteTopic;
+		}
 	}
 
+	onChangeTopic = event => this.setState( { siteTopicInputValue: event.target.value } );
```

Comment:
Maybe we should trim the value? Like so:  `trim( event.target.value )`. This is because if I give only spaces in the field, I will still get the continue button:
<img width="779" alt="screenshot 2018-11-01 at 10 37 34 am" src="https://user-images.githubusercontent.com/1269602/47833685-30632d00-ddc2-11e8-9121-804e2e133c47.png">